src/lib/storage/reference_column/reference_column_iterable.hpp

```diff
@@ -60,7 +60,7 @@ class ReferenceColumnIterable : public ColumnIterable<ReferenceColumnIterable<T>
       const auto chunk_id = _pos_list_it->chunk_id;
       const auto& chunk_offset = _pos_list_it->chunk_offset;
 
-      if (chunk_id != _cached_chunk_id || !_cached_column) {
+      if (chunk_id != _cached_chunk_id) {
```

Comment:
@Bensk1 : do you think that if could be predicated? I haven't thought enough about predication to fully understand it. :/

Reply:
Well, probably doesn't matter as the branch pred. will be right in 99.9999% of the cases? Right?

Reply:
Correct, it does not make sense here and either the branch prediction should be almost always right or the there should be only very few offsets/chunks.